src/main/java/bio/terra/cli/context/Resource.java

```diff
@@ -0,0 +1,295 @@
+package bio.terra.cli.context;
+
+import bio.terra.cli.command.exception.UserActionableException;
+import bio.terra.cli.context.resources.AiNotebook;
+import bio.terra.cli.context.resources.BqDataset;
+import bio.terra.cli.context.resources.GcsBucket;
+import bio.terra.cli.context.utils.Printer;
+import bio.terra.cli.service.WorkspaceManagerService;
+import bio.terra.workspace.model.AccessScope;
+import bio.terra.workspace.model.CloningInstructionsEnum;
+import bio.terra.workspace.model.ControlledResourceIamRole;
+import bio.terra.workspace.model.ControlledResourceMetadata;
+import bio.terra.workspace.model.ManagedBy;
+import bio.terra.workspace.model.PrivateResourceUser;
+import bio.terra.workspace.model.ResourceDescription;
+import bio.terra.workspace.model.ResourceMetadata;
+import bio.terra.workspace.model.ResourceType;
+import bio.terra.workspace.model.StewardshipType;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+import com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder;
+import java.io.PrintStream;
+import java.util.List;
+import java.util.UUID;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+/**
+ * POJO class that represents a workspace resource. This class is serialized to disk as part of the
+ * global context. It is also used as the user-facing JSON output for commands that return a
+ * resource.
```

Comment:
That's a good point about decoupling serialization and our internal object representation. For now, I think it's fine to couple our serialization format with the command output JSON format, but not having to think about backwards compatibility every time we want to change e.g. the `Resource` class sounds more important.

I'll take a pass at separating this out in this PR and if it gets too unwieldy will do it as a follow-on.

Reply:
To be clear, I'm ok letting this POR go in as now for serialization and changing anything about serialization as a follow up ticket, but totally up to you.

Reply:
Yeah, I'm not sure it's worth including here. But I'm thinking it might cause a fairly large reorg and want to see what that would look like before going ahead with this restructuring.